storm-webapp/src/main/java/org/apache/storm/daemon/ui/UIHelpers.java

```diff
@@ -1312,21 +1326,27 @@ private static Double nullToZero(Double value) {
         Map<String, Object> result = new HashMap();
         ExecutorSummary executorSummary = executorAggregateStats.get_exec_summary();
         ExecutorInfo executorInfo = executorSummary.get_executor_info();
-        ComponentAggregateStats componentAggregateStats = executorAggregateStats.get_stats();
-        SpecificAggregateStats specificAggregateStats = componentAggregateStats.get_specific_stats();
-        BoltAggregateStats boltAggregateStats = specificAggregateStats.get_bolt();
-        CommonAggregateStats commonAggregateStats = componentAggregateStats.get_common_stats();
         String executorId = prettyExecutorInfo(executorInfo);
         result.put("id", executorId);
-        result.put("encodedId", URLEncoder.encode(executorId));
+        try {
+            result.put("encodedId", URLEncoder.encode(executorId, StandardCharsets.UTF_8.name()));
+        } catch (UnsupportedEncodingException e) {
+            throw Utils.wrapInRuntime(e);
+        }
         result.put("uptime", prettyUptimeSec(executorSummary.get_uptime_secs()));
         result.put("uptimeSeconds", executorSummary.get_uptime_secs());
         String host = executorSummary.get_host();
         result.put("host", host);
         int port = executorSummary.get_port();
         result.put("port", port);
+        
+        ComponentAggregateStats componentAggregateStats = executorAggregateStats.get_stats();
```

Comment:
Are below changes relevant / necessary? Because it looks like not relevant to this issue.

Reply:
Checkstyle started complaining that the variable declarations were too far from first use.

Reply:
Ah yes thanks for fixing this too!